code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -89,7 +113,36 @@ private Token createToken(ApiRequest request) {
         }
         token.setIdentities(transFormedIdentities);
         token.setUserIdentity(identityManager.untransform(token.getUserIdentity(), true));
-        token.setJwt(authTokenDao.createToken(token.getJwt(), token.getAuthProvider(), ((Policy) ApiContext.getContext().getPolicy()).getAccountId()).getKey());
+        String dbJwt = token.getJwt();
+        try {
+            jsonData = tokenService.getJsonPayload(dbJwt, true);
+        } catch (TokenException e) {
+            throw new ClientVisibleException(ResponseCodes.FORBIDDEN);
+        }
+
+        if (jsonData == null) {
+            throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, AbstractTokenUtil.ACCESS_TOKEN_INVALID,
+                    "Json Web Token invalid.", null);
+        }
+
+        String accountId = ObjectUtils.toString(jsonData.get(AbstractTokenUtil.ACCOUNT_ID), null);
+        if (accountId == null) {
+            return null;
+        }
+        TokenUtil tokenUtil = tokenAuthLookup.tokenUtils();
+        Account account = authDao.getAccountByExternalId(accountId, tokenUtil.userType());
+        if (account != null) {
+            authenticatedAsAccountId = account.getId();
+        }
+        long tokenAccountId = (accountDao.findByUuid(AbstractTokenUtil.TOKEN)).getId();
+
```

Comment:
I dont think you should get the accountId of kind "token" from accountDao this way.

Use ((Policy) ApiContext.getContext().getPolicy()).getAccountId() - that is what the code used while creating the token. This should resolve to "token" accountId.